bigtable-dataflow-parent/bigtable-beam-import/README.md

```diff
@@ -1,44 +1,215 @@
+    cbt createfamily your-table-name your-column-family
+
+Once your import is completed follow the instructions for the validator below to ensure it was successful.
+
+Please pay attention to the Cluster CPU usage and adjust the number of Dataflow workers accordingly.
+
+### Snapshots (preferred method)
+
+1. Set the environment variables.
+    ```
+    PROJECT_ID=your-project-id
+    INSTANCE_ID=your-instance-id
+    TABLE_NAME=your-table-name
+    REGION=us-central1
+
+    SNAPSHOT_GCS_PATH="$BUCKET_NAME/hbase-migration-snap"
+    SNAPSHOT_NAME=your-snapshot-name
+    ```
+    
+1. Run the import.
+    ```
+    java -jar bigtable-beam-import-1.14.1-SNAPSHOT-shaded.jar importsnapshot \
+        --runner=DataflowRunner \
+        --project=$PROJECT_ID \
+        --bigtableInstanceId=$INSTANCE_ID \
+        --bigtableTableId=$TABLE_NAME \
+        --hbaseSnapshotSourceDir=$SNAPSHOT_GCS_PATH/data \
+        --snapshotName=$SNAPSHOT_NAME \
+        --stagingLocation=$SNAPSHOT_GCS_PATH/staging \
+        --tempLocation=$SNAPSHOT_GCS_PATH/temp \
+        --region=$REGION
+    ```
+
+
+### Sequence Files
+
+1. Set the environment variables.
+    ```
+    PROJECT_ID=your-project-id
+    INSTANCE_ID=your-instance-id
+    CLUSTER_NUM_NODES=3
+    CLUSTER_ZONE=us-central1-a
+    TABLE_NAME=your-table-name
+    
+    BUCKET_NAME=gs://bucket-name
+    ```
+1. Run the import.
+    ```
+    java -jar bigtable-beam-import-1.14.1-shaded.jar import \
+        --runner=dataflow \
+        --project=$PROJECT_ID \
+        --bigtableInstanceId=$INSTANCE_D \
+        --bigtableTableId=$TABLE_NAME \
+        --sourcePattern='$BUCKET_NAME/hbase-export/part-*' \
+        --tempLocation=$BUCKET_NAME/hbase_temp \
+        --maxNumWorkers=$(expr 3 \* $CLUSTER_NUM_NODES)  \
+        --zone=$CLUSTER_ZONE
+    ```
+
+
+## Validating data
+
+Once your snapshot or sequence file is imported, you should run the validator to
+check if there are any rows with mismatched data. 
+
+1. Set the environment variables.
+    ```
+    PROJECT_ID=your-project-id
+    INSTANCE_ID=your-instance-id
+    TABLE_NAME=your-table-name
+    REGION=us-central1
+    
+    SNAPSHOT_GCS_PATH="$BUCKET_NAME/hbase-migration-snap"
+    ```
+1. Run the sync job. It will put the results into `$SNAPSHOT_GCS_PATH/data-verification/output`,
+so if you run the command multiple times, you might want to add a number to the output,
+like so `$SNAPSHOT_GCS_PATH/data-verification/output-1`. 
+    ```
+    java -jar bigtable-beam-import-1.14.1-SNAPSHOT-shaded.jar sync-table  \
+        --runner=dataflow \
+        --project=$PROJECT_ID \
+        --bigtableInstanceId=$INSTANCE_D \
+        --bigtableTableId=$TABLE_NAME \
+        --outputPrefix=$SNAPSHOT_GCS_PATH/sync-table/output \
+        --stagingLocation=$SNAPSHOT_GCS_PATH/sync-table/staging \
+        --hashTableOutputDir=$SNAPSHOT_GCS_PATH/hashtable \
+        --tempLocation=$SNAPSHOT_GCS_PATH/sync-table/dataflow-test/temp \
+        --region=$REGION
+    ```
```

Comment:
any cleanup needed?

Reply:
Output files are generated but the customer should determine their life-cycle and clean them up when migration is complete.